### PR TITLE
Turki Almalki

### PR DESCRIPTION
**Pull request by Turkialmalki**

### CODE REVIEW

File: homwork.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  /////////////////////////Q2////////////////////////////////
+  function even_digits(num) {
+    var ctr = 0;
+    while (num) {
+      ctr += num % 2 === 0;
+      num = Math.floor(num / 10);
+    }
+    return ctr;
+  }
```
Comment (alanoud-GA) [inline homwork.js:18]: The questions require a function that will take a number as the end of a range and sums all the even numbers from 1 to the end range number
so if we call `even_digits(5)` it should add 2 and 4 as they are the only even numbers between 1 and 5. The output would be 6.

File: homwork.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    return stringRev;
+}
+alert(reverseString("Hello World!"));
+
+///////////////////////////////Q5////////////////////////////////
+function combineInto(a,b) {
+    var len = a.length;
+    for (var i=0; i < len; i=i+5000) {
+        b.unshift.apply( b, a.slice( i, i+5000 ) );
+    }
```
Comment (alanoud-GA) [inline homwork.js:50]: Can you explain this solution?

File: homwork.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    var s = "";
+    for(var i = 1; i <= num; i++) {
+      s += i;
+  
+      if (i < (num)) {
+        s += ', ';
+      }
+    }
+    return s;
+  }
```
Comment (alanoud-GA) [inline homwork.js:64]: This counts up but doesn't count down

File: homwork.js (hunk 1 above)
Comment (Turkialmalki) [reply homwork.js:18]: Yes, I need to update the answer with the code of the following snippet:
let x = 1;
let y = 2;
let sum = 0;
let limit = 10000;
let evensum = 2;

while ((x + y) < limit) {
    sum = x + y;
    x = y;
    y = sum;

    if (sum % 2 === 0) {
        evensum += sum;
    }
    console.log(evensum);
}

File: homwork.js (hunk 2 above)
Comment (Turkialmalki) [reply homwork.js:50]: So, passing either a or b to apply(..)'s second argument, means that the array is being spread out as arguments to the function. 
if the array being added on has a million items in it, you'd almost certainly way exceed the size of the stack allowed for that unshift(..) call. 

And I batch up segments at the max safe size which is 5000

File: homwork.js (hunk 3 above)
Comment (Turkialmalki) [reply homwork.js:64]: Yes, here's the function using recursion method for the count down where a function calls itself:
function countdown(i)   
{
    if(i<=0)    
        return i;
    return i + " " + countdown(--i);
}

**Pull request closed without merging**